Source/DafnyCore/DafnyPrelude.bpl

```diff
@@ -572,9 +565,9 @@ const unique allocName: NameFamily;
 // -- Arrays -----------------------------------------------------
 // ---------------------------------------------------------------
 
-function _System.array.Length(a: ref): int uses {
-  axiom (forall o: ref :: 0 <= _System.array.Length(o));
-}
+function _System.array.Length(a: ref): int;
+axiom (forall o: ref :: {_System.array.Length(o)} 0 <= _System.array.Length(o));
```

Comment:
I added a trigger here. Alternatively the trigger can be removed and the axiom moved back into the `uses` clause, if someone thinks this is a bad idea. `_System.array.Length` does not appear in any other triggers, so adding the trigger should not change the semantics I think.

Reply:
How does the trigger affect the SMT solving? Was there a chance it would previously instantiate the quantifier even though there was no `_System.array.Length` call ?

Reply:
Z3 infers patterns if none are provided. In this case this is trivial, it would be the same pattern I manually added. This change should have no affect on SMT solving.